lib/system.nim

```diff
@@ -4431,3 +4431,19 @@ when defined(genode):
       componentConstructHook(env)
         # Perform application initialization
         # and return to thread entrypoint.
+
+proc name(t: typedesc): string {.magic: "TypeTrait".}
+  ## pending https://github.com/nim-lang/Nim/issues/10078, nest inside
+  ## `$`*(t: typedesc)
+
+template `$`*(t: typedesc): string =
```

Comment:
Why can't this be a `proc`? Doesn't the `typedesc` parameter already make it a compile time evaluation?

Reply:
This should be a ``proc`` and ``semmagic.nim`` line 139 needs to be updated to read like ``of "name", "$"``.

Reply:
done